py/dml/dmlc.py

```diff
@@ -281,102 +300,98 @@ def main(argv):
     # <dt>-I <i>path</i></dt>
     # <dd>Add <arg>path</arg> to the search path for imported
     # modules.</dd>
-    optpar.add_option(
-        '-I', dest = 'import_path', action = 'append',
-        metavar = 'PATH',
-        default = [],
-        help = 'add PATH to import search path')
+    parser.add_argument(
+        '-I', dest='import_path', action='append',
+        metavar='PATH',
+        default=[],
+        help='add PATH to import search path')
 
     # <dt>-D <i>name</i>=<i>definition</i></dt>
     # <dd>Define a compile-time parameter.  The definition
     # must be a literal expression, and can be a quoted
     # string, a boolean, an integer, or a floating point
     # constant. The parameter will appear in the top-level scope.</dd>
-    optpar.add_option(
-        '-D', dest = 'defines', action = 'append',
-        metavar = 'NAME=VALUE',
-        default = [],
-        help = 'set compile time constant NAME to VALUE')
+    parser.add_argument(
+        '-D', dest='defines', action='append',
+        metavar='NAME=VALUE',
+        default=[],
+        help='set compile time constant NAME to VALUE')
 
     # <dt>--dep</dt>
     # <dd>Output makefile rules describing dependencies.</dd>
-    optpar.add_option(
-        '--dep', dest = "makedep", action = 'store',
-        help = 'generate makefile dependencies')
+    parser.add_argument(
+        '--dep', action='store',
+        help='generate makefile dependencies')
 
     # <dt>--no-dep-phony</dt>
     # <dd>With --dep, avoid addition of a phony target for each dependency
     # other than the main file.</dd>
-    optpar.add_option(
-        '--no-dep-phony', dest = "no_dep_phony", action = 'store_true',
-        help = 'With --dep, avoid addition of a phony target for each'
+    parser.add_argument(
+        '--no-dep-phony', action='store_true',
```

Comment:
So with no explicit `dest` the parser will parse `--no-dep-phony`, and turn it into `no_dep_phony` when genning the options? Huh.

Reply:
Yes, that's standard practice for argparse. Adn `add_argument('-f', '--foo-bar', ...)` stores it in `foo_bar`.